File: src/dataset/Java/armstrong-numbers_exercism-java-uzilan_robiworks/openAI/ChatGPT4_armstrong-numbers_v1.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int sum = 0, temp = numberToCheck, exp = (int) Math.log10(numberToCheck) + 1;
        while (temp > 0) {
            int digit = temp % 10;
            sum += fastPower(digit, exp);
            temp /= 10;
        }
        return sum == numberToCheck;
    }

    int fastPower(int base, int exp) {
        int result = 1;
        while (exp > 0) {
            if ((exp & 1) == 1) {
                result *= base;
            }
            base *= base;
            exp >>= 1;
        }
        return result;
    }

}

File: src/dataset/typescript/armstrong-numbers_Exercism-typescript-shybyte/openAI/ChatGPT4_armstrong-numbers_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function sum(numbers) {
    let total = 0;
    for (const num of numbers) {
        total += num;
    }
    return total;
}
class ArmstrongNumbers {
    static isArmstrongNumber(x) {
        const digits = x.toString();
        const length = digits.length;
        let armstrongSum = 0;
        for (let i = 0; i < length; i++) {
            const digit = digits.charCodeAt(i) - 48; // Faster than parseInt
            armstrongSum += Math.pow(digit, length);
            if (armstrongSum > x)
                return false; // Early exit if sum exceeds x
        }
        return armstrongSum === x;
    }
}
exports.default = ArmstrongNumbers;
